storagenode/piecestore/endpoint.go

```diff
@@ -166,6 +168,42 @@ func (endpoint *Endpoint) Delete(ctx context.Context, delete *pb.PieceDeleteRequ
 	return &pb.PieceDeleteResponse{}, nil
 }
 
+// DeletePiece handles deleting a piece on piece store requested by satellite.
+//
+// It doesn't return an error if the piece isn't found by any reason.
+func (endpoint *Endpoint) DeletePiece(
+	ctx context.Context, req *pb.DeletePieceRequest,
+) (_ *pb.DeletePieceResponse, err error) {
+	defer mon.Task()(&ctx, req.PieceId.String())(&err)
+
+	peer, err := identity.PeerIdentityFromContext(ctx)
+	if err != nil {
+		return nil, rpcstatus.Error(rpcstatus.Unauthenticated, Error.Wrap(err).Error())
+	}
+
+	err = endpoint.trust.VerifySatelliteID(ctx, peer.ID)
+	if err != nil {
+		return nil, rpcstatus.Error(rpcstatus.PermissionDenied,
+			Error.New("%s", "delete piece called with untrusted ID").Error(),
+		)
+	}
+
+	err = endpoint.store.Delete(ctx, peer.ID, req.PieceId)
+	if err != nil {
+		endpoint.log.Error("delete piece failed",
+			zap.Error(Error.Wrap(err)),
+			zap.Stringer("Satellite ID", peer.ID),
+			zap.Stringer("Piece ID", req.PieceId),
+		)
+
+		return nil, rpcstatus.Error(rpcstatus.Internal,
+			Error.New("%s", "delete piece failed").Error(),
+		)
+	}
```

Comment:
Should we recognize the NotFound case (if `peer.ID` and/or `req.PieceID` are not found) and return the appropriate rpc error for it instead of `Internal`?

Reply:
See my comment above and this PR as a reference https://github.com/storj/storj/pull/3591

Reply:
ok

Reply:
As commented in the stand up I was wrong and the method returns errors for a not found piece however it doesn't return a different error class than other errors so the current workaround is to parse the error string.

Because I discovered it when I've realized that I missed yesterday to inspect the used implementation (difficult to see sometimes because we use interfaces types), I updated https://github.com/storj/storj/pull/3591

Also, I've created a ticket for fixing the implementation and have different error class for not found piece at some point https://storjlabs.atlassian.net/browse/V3-3222